src/transaction/sign.ts

```diff
@@ -55,12 +49,61 @@ function signWithKeypair(
   }
 
   const serialized = binary.encode(tx)
+
+  // Decode the serialized transaction:
+  const decoded = binary.decode(serialized)
+
+  // ...And ensure it is equal to the original tx, except:
+  // - It must have a TxnSignature or txSigners (multisign).
```

Comment:
nit: Refer to this property the same way you refer to it in code: 'Signers' vs txSigners, or at the very least 'tx.Signers'.

Or, if you think it should be called txSigners, you could refactor the code to this property name.

Reply:
Not done